float64[] gain_matrix
float64[] center
float64[] control
float64[] cost
float64 radius